akka-projection-grpc/src/main/scala/akka/persistence/EventWriter.scala

```diff
@@ -0,0 +1,339 @@
+              }
+            }
+          }
+
+          def handleJournalResponse(response: JournalProtocol.Response): Behavior[AnyRef] =
+            response match {
+              case JournalProtocol.WriteMessageSuccess(message, `actorInstanceId`) =>
+                val pid = message.persistenceId
+                val sequenceNr = message.sequenceNr
+                perPidWriteState.get(pid) match {
+                  case None =>
+                    throw new IllegalStateException(
+                      s"Got write success reply for event with no waiting request, probably a bug (pid $pid, seq nr $sequenceNr)")
+                  case Some(stateForPid) =>
+                    stateForPid.waitingForReply.get(sequenceNr) match {
+                      case None =>
+                        throw new IllegalStateException(
+                          s"Got write success reply for event with no waiting request, probably a bug (pid $pid, seq nr $sequenceNr)")
+                      case Some((_, waiting)) =>
+                        if (context.log.isTraceEnabled)
+                          context.log.trace2(
+                            "Successfully wrote event persistence id [{}], sequence nr [{}]",
+                            pid,
+                            message.sequenceNr)
+                        waiting ! StatusReply.success(WriteAck(pid, sequenceNr))
+                        val newState = stateForPid.copy(waitingForReply = stateForPid.waitingForReply - sequenceNr)
+                        handleUpdatedStateForPid(pid, newState)
+                        Behaviors.same
+                    }
+                }
+
+              case JournalProtocol.WriteMessageFailure(message, error, `actorInstanceId`) =>
+                val pid = message.persistenceId
+                val sequenceNr = message.sequenceNr
+                perPidWriteState.get(pid) match {
+                  case None =>
+                    throw new IllegalStateException(
+                      s"Got error reply for event with no waiting request, probably a bug (pid $pid, seq nr $sequenceNr)",
+                      error)
+                  case Some(state) =>
+                    // write failure could be re-delivery, we need to check
+                    state.waitingForReply.get(sequenceNr) match {
+                      case None =>
+                        throw new IllegalStateException(
+                          s"Got error reply for event with no waiting request, probably a bug (pid $pid, seq nr $sequenceNr)",
+                          error)
+                      case Some(_) =>
+                        // quite likely a re-delivery of already persisted events, check highest seqnr
+                        val sortedSeqNrs = state.waitingForReply.keys.toSeq.sorted
+                        val maxSeqNrFinderName = URLEncoder.encode(s"MaxSeqNrFinder-$pid", "UTF-8")
+                        if (context.child(maxSeqNrFinderName).isEmpty) {
```

Comment:
Could just be an anonymous actor, since it's stopping when it's done.

Maybe we don't need an actor? With my suggestion it will only be one reply `RecoverySuccess(highestSequenceNr)` and we can use `context.ask` instead.

Reply:
Won't we see that highest event replayed though, or would it be something like replay from=to=Long.MaxValue to not get any replayed events?